addons/l10n_ro_efactura/controllers/main.py

```diff
@@ -0,0 +1,61 @@
+import requests
+from odoo import _, http
+from odoo.exceptions import UserError
+from odoo.http import request
+from werkzeug.urls import url_join, url_encode
+
+
+ENDPOINT_AUTHORIZE = 'https://logincert.anaf.ro/anaf-oauth2/v1/authorize'
+ENDPOINT_TOKEN = 'https://logincert.anaf.ro/anaf-oauth2/v1/token'
+
+
+class EFacturaOAuthController(http.Controller):
+
+    @http.route('/l10n_ro_edi/authorize/<int:company_id>', auth="user")
+    def authorize(self, company_id, **kw):
+        """ Generate Authorization Token to acquire access_key for requesting Access Token """
+        company = http.request.env['res.company'].sudo().browse(company_id)
+        if not company.l10n_ro_edi_client_id or not company.l10n_ro_edi_client_secret:
+            raise UserError(_("Client ID and Client Secret field must be filled."))
+
+        auth_url = url_join(ENDPOINT_AUTHORIZE, '?' + url_encode({
+            'response_type': 'code',
+            'client_id': company.l10n_ro_edi_client_id,
+            'redirect_uri': company.l10n_ro_edi_callback_url,
```

Comment:
I think there is a security issue here. `authorize` is public and browse the company in `sudo` mode.